business/mesh.go

```diff
@@ -462,7 +459,6 @@ func (in *MeshService) resolveKialiNetwork() (string, error) {
 // visible to the adjacent mesh control plane. This assumes that the Istio namespace is
 // named the same as in Kiali's Cluster.
 func (in *MeshService) resolveRemoteClustersFromSecrets() ([]Cluster, error) {
```

Comment:
Perhaps this can be a separate PR but imo we should move this logic to the client factory and attach the cluster info to the kiali `kubernetes.ClientInterface`. Right now there's a few different places that call `ClientConfig()` to recreate the Kiali Service Account client but we only need to create this client once and then pass off the client to the various business services. Managing the client creation in a single place will also greatly simplify using clients for mutlicluster.

So if all the logic for getting the cluster info from the secrets and turning those into `kubernetes.ClientInterface` goes into the client factory and if something like a `GetClusterInfo` method gets added to `kubernetes.ClientInterface`, then the mesh service would just call `GetClusterInfo()` for each one of the Kiali Service Account clients.

But again maybe this change should just focus on reading the secrets from fs rather than from the kube API and another change can refactor this logic into the client factory. Either way is fine.

Reply:
> But again maybe this change should just focus on reading the secrets from fs rather than from the kube API and another change can refactor this logic into the client factory. Either way is fine.

Right. That is the intent. We'll iterate over this is different PRs.